django/db/backends/postgresql/base.py

```diff
@@ -189,7 +189,7 @@ def ensure_timezone(self):
         conn_timezone_name = self.connection.get_parameter_status('TimeZone')
         timezone_name = self.timezone_name
         if timezone_name and conn_timezone_name != timezone_name:
-            with self.connection.cursor() as cursor:
+            with self.cursor() as cursor:
```

Comment:
I think that you should partly restore behavior before https://github.com/django/django/commit/b484f167bed61f4cff215208eddf98a0655239d4 i.e.
```python
cursor = self.connection.cursor()
try:
    cursor.execute(self.ops.set_time_zone_sql(), [self.timezone_name])
finally:
    cursor.close()
```
instead of
```python
with self.connection.cursor() as cursor:
    cursor.execute(self.ops.set_time_zone_sql(), [timezone_name])
```